### PR TITLE
Test and rework of filterBoolean

### PR DESCRIPTION
**Pull request by GGalya1**
- null and undefined checking has been removed, since TypeScript does not allow null or undefinded to be written to a field of type boolean (enabled field in tinkerforgeDTO).
- added test case for both valid outputs and for two edge cases (when the output is 'false')

### CODE REVIEW

**Review by Christina-Ev — DISMISSED** [review]

File: tests/usePosts.test.ts [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+  test("should return true for 'on', 'enabled', 'true' when post.enabled is true", () => {
+    const post: tinkerforgeDTO = { enabled: true } as tinkerforgeDTO;
+
+    expect(filterBoolean("on", post)).toBe(true);
+    expect(filterBoolean("enabled", post)).toBe(true);
+    expect(filterBoolean("true", post)).toBe(true);
+  });
+
+  test("should return true for 'off', 'disabled', 'false' when post.enabled is false", () => {
+    const post: tinkerforgeDTO = { enabled: false } as tinkerforgeDTO;
```
Comment (Christina-Ev) [inline tests/usePosts.test.ts:17]: as tinkerforgeDTO; seems unnecessary since the test object already matches the structure

File: tests/usePosts.test.ts [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+
+    expect(filterBoolean("off", post)).toBe(true);
+    expect(filterBoolean("disabled", post)).toBe(true);
+    expect(filterBoolean("false", post)).toBe(true);
+  });
+
+  test("should return false for unsupported query values", () => {
+    const post: tinkerforgeDTO = { enabled: true } as tinkerforgeDTO;
```
Comment (Christina-Ev) [inline tests/usePosts.test.ts:25]: as tinkerforgeDTO; 
could be removed

File: tests/usePosts.test.ts [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+
+    expect(filterBoolean("yes", post)).toBe(false);
+    expect(filterBoolean("no", post)).toBe(false);
+    expect(filterBoolean("randomText", post)).toBe(false);
+  });
+
+  test("should return false if the query matches but the post.enabled state does not", () => {
+    const postTrue: tinkerforgeDTO = { enabled: true } as tinkerforgeDTO;
```
Comment (Christina-Ev) [inline tests/usePosts.test.ts:33]: as tinkerforgeDTO; could be removed

File: tests/usePosts.test.ts [hunk 4]
```diff
@@ -1,6 +1,43 @@
...
+    const postFalse: tinkerforgeDTO = { enabled: false } as tinkerforgeDTO;
```
Comment (Christina-Ev) [inline tests/usePosts.test.ts:34]: as tinkerforgeDTO; could be removed

File: tests/usePosts.test.ts (hunk 1 above)
Comment (GGalya1) [reply tests/usePosts.test.ts:17]: TypeScript frowns on the lack of a type. It is also easier to create objects this way

File: tests/usePosts.test.ts [hunk 5]
```diff
@@ -1,6 +1,43 @@
...
+describe("filterBoolean", () => {
+  test("should return true for 'on', 'enabled', 'true' when post.enabled is true", () => {
+    const post: tinkerforgeDTO = { enabled: true } as tinkerforgeDTO;
+
+    expect(filterBoolean("on", post)).toBe(true);
+    expect(filterBoolean("enabled", post)).toBe(true);
+    expect(filterBoolean("true", post)).toBe(true);
+  });
+
+  test("should return true for 'off', 'disabled', 'false' when post.enabled is false", () => {
```
Comment (Christina-Ev) [inline tests/usePosts.test.ts:16]: You could update the title of the test, to better match the function's behaviour.
test("should return true when the query matches the boolean state of post.enabled"

**Review by Christina-Ev — CHANGES_REQUESTED** [review]

File: tests/usePosts.test.ts (hunk 2 above)
Comment (GGalya1) [reply tests/usePosts.test.ts:25]: Type '{ enabled: true; }' is missing the following properties from type '{ id: string; date_created: string; date_modified: string; enabled: boolean; description: string | null; uid: number; config: Record<string, { description: string | null; interval: number; trigger_only_on_change: boolean; topic: string; unit: string; }>; on_connect: { ...; }[]; label?: string | ... 1 more ... | unde...': id, date_created, date_modified, description, and 3 more.

File: tests/usePosts.test.ts (hunk 3 above)
Comment (GGalya1) [reply tests/usePosts.test.ts:33]: see above

File: tests/usePosts.test.ts (hunk 4 above)
Comment (GGalya1) [reply tests/usePosts.test.ts:34]: we need it to create an object of type tinkerforgeDTO not just a value

File: tests/usePosts.test.ts (hunk 5 above)
Comment (GGalya1) [reply tests/usePosts.test.ts:16]: It seems that the current description better specifies under what inputs what result is expected, without the need to study the implementation of the tested method itself

File: tests/usePosts.test.ts (hunk 2 above)
Comment (Christina-Ev) [reply tests/usePosts.test.ts:25]: Sorry, forgot to add the rest of the comments. In my case it it was fixed by mocking, for example, instead of using the line (const post: tinkerforgeDTO = { enabled: true } as tinkerforgeDTO;).
const post: tinkerforgeDTO = {
      enabled: false,
      id: "",
      date_created: "",
      date_modified: "",
      description: null,
      uid: 0,
      config: {},  
      on_connect: []
    };
Since this is a simple test, I suppose mocking is not that necessary, considering it only checks a siple assignment. 
Sorry for not adding that earlier

Comment (Christina-Ev) [reply tests/usePosts.test.ts:25]: Just ignore my previous comments on that

**Review by Christina-Ev — APPROVED** [review]
